tinygrad/codegen/uopgraph.py

```diff
@@ -239,6 +239,9 @@ def reduce_collapse(acc:UOp, ret:UOp, alu:UOp):
 arange_augrng = UPat.any(rng_aug, rng_aug+UPat.var("idx2"), rng_aug+UPat.var("idx2")+UPat.var("idx3"), UPat(Ops.VECTORIZE, name="vec", src=rng_aug))
 arange_m = ((arange_augrng<UPat.cvar("compval"))!=UPat(Ops.CONST, name="ne", arg=True)).where(UPat.cvar("multconst"), UPat.const(None, 0))
 
+# this moves the accumulation variable down an unrolled add chain which allows for more efficient accumulation using mulacc
+mulacc_unrolled = PatternMatcher([(UPat.var("x")+UPat.var("y")+acc_pat, lambda x,y,acc: (acc+x)+y if y.op is not Ops.DEFINE_ACC else None)])
```

Comment:
does it need its own pass?

Reply:
I can't add it to sym since it will break loop/index/reduce_collapse, I tried adding it to `late_reduce` but since that will rewrite from the sources some the mul/adds will have been changed to mulacc and the `DEFINE_ACC` can't be moved down the chain anymore.

Reply:
I can add it to the devectorize + load_store_indexing pass but that might be less efficient